File: BinarySearch.cpp
int binarySearch(int *input, int n, int val)
{
    int start = 0; int end = n-1; int mid = (start + end)/2;
    while(start <= end) {
        if(input[mid] == val)
            return mid;
        else if (val > input[mid]) {
			start = mid+1;
            mid = (start + end)/2;
        }
        else {
            end = mid-1;
            mid = (start + end)/2;
        }
	}
    return -1;
}

File: Binarysearch.py
# Binary Search in python

def binarySearch(array, x, low, high):

    if high >= low:

        mid = low + (high - low)//2

      

        if array[mid] == x:

            return mid

        

        elif array[mid] > x:

            return binarySearch(array, x, low, mid-1)

        

        else:

            return binarySearch(array, x, mid + 1, high)

    else:

        return -1

array = [3, 4, 5, 6, 7, 8, 9]

x = 4

result = binarySearch(array, x, 0, len(array)-1)

if result != -1:

    print("Element is present at index " + str(result))

else:

    print("Not found")
